### PR TITLE
test upgrade for monitoring components

### PR DESCRIPTION
**Pull request by k15r**
**Description**

Changes proposed in this pull request:

- Upgrade tests for prometheus and grafana

**Related issue(s)**
#2951

### CODE REVIEW

File: tests/end-to-end/upgrade/pkg/tests/monitoring/metrics.go [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+	time := time.Now().Add(-time.Minute)
+	log.Debugln("before collect")
+	result, err := ut.collectMetrics(time)
+	if err != nil {
+		return err
+	}
+	log.Debugln("after collect")
+	log.Debugln("before store")
+	err = ut.storeMetrics(result, time)
+	log.Debugln("after store")
```
Comment (rakesh-garimella) [inline tests/end-to-end/upgrade/pkg/tests/monitoring/metrics.go]: Do you want these debug messages.

File: tests/end-to-end/upgrade/pkg/tests/monitoring/metrics.go [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+	}
+	return timeValue, value, nil
+}
+
+func (ut *MetricsUpgradeTest) compareMetrics() error {
+	time, previous, err := ut.retrievePreviousMetrics()
+	if err != nil {
+		return err
+	}
+	current, err := ut.collectMetrics(time)
```
Comment (polskikiel) [inline tests/end-to-end/upgrade/pkg/tests/monitoring/metrics.go]: check this error and return it if `err != nil`

File: tests/end-to-end/upgrade/pkg/tests/monitoring/grafana.go [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+	if err != nil {
+		return err
+	}
+
+	dashboards, err := ut.collectDashboards()
+	if err != nil {
+		return err
+	}
+
+	err = ut.storeDashboards(dashboards)
```
Comment (polskikiel) [inline tests/end-to-end/upgrade/pkg/tests/monitoring/grafana.go]: just return ut.storeDashboards(dashboards)

File: tests/end-to-end/upgrade/pkg/tests/monitoring/grafana.go [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+
+const (
+	grafanaDataField           = "dashboards"
+	grafanaAdminUserSecretName = "admin-user"
+	grafanaConfigMapName       = "grafana-upgrade-test"
+	grafanaNS                  = "kyma-system"
+	grafanaPodName             = "monitoring-grafana-0"
+	grafanaServiceName         = "monitoring-grafana"
+	grafanaStatefulsetName     = "monitoring-grafana"
+	grafanaPvcName             = "monitoring-grafana"
```
Comment (polskikiel) [inline tests/end-to-end/upgrade/pkg/tests/monitoring/grafana.go]: ^unused fields 
do we need them?

File: tests/end-to-end/upgrade/pkg/tests/monitoring/grafana.go [hunk 5]
```diff
@@ -0,0 +1,341 @@
...
+		grafana: grafana{loginForm: url.Values{}},
+	}
+}
+
+// CreateResources retrieves all installed dashboards and stores this information in an config map
+func (ut *GrafanaUpgradeTest) CreateResources(stop <-chan struct{}, log logrus.FieldLogger, namespace string) error {
+	ut.namespace = namespace
+	ut.log = log
+	err := ut.getGrafana()
+	if err != nil {
```
Comment (polskikiel) [inline tests/end-to-end/upgrade/pkg/tests/monitoring/grafana.go]: you can make `if err := ut.getGrafana(); err != nil{
}`

File: tests/end-to-end/upgrade/pkg/tests/monitoring/metrics.go (hunk 1 above)
Comment (polskikiel) [reply tests/end-to-end/upgrade/pkg/tests/monitoring/metrics.go]: ?

Comment (polskikiel) [reply tests/end-to-end/upgrade/pkg/tests/monitoring/metrics.go]: ?

Comment (k15r) [reply tests/end-to-end/upgrade/pkg/tests/monitoring/metrics.go]: @rakesh-garimella no ;)

**Pull request merged**